chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/config.rs

```diff
@@ -1,34 +1,16 @@
 use crate::state::SourceChainConfig;
 
-pub fn get_on_ramps(config: &SourceChainConfig) -> Vec<&[u8]> {
-    let first_item = trim_trailing_zeros(&config.on_ramp[0]);
-    let second_item = trim_trailing_zeros(&config.on_ramp[1]);
-
-    let mut on_ramps: Vec<&[u8]> = Vec::new();
-
-    // filter out empty values
-    if !first_item.is_empty() {
-        on_ramps.push(first_item);
-    }
-    if !second_item.is_empty() {
-        on_ramps.push(second_item);
-    }
-
-    on_ramps
+pub fn get_on_ramps(config: &SourceChainConfig) -> impl Iterator<Item = &[u8; 64]> {
+    config.on_ramp.iter().filter(|i| *i != &[0u8; 64])
 }
 
 pub fn is_on_ramp_configured(config: &SourceChainConfig, on_ramp: &[u8]) -> bool {
-    let valid_on_ramps = get_on_ramps(config);
-
-    valid_on_ramps.contains(&on_ramp)
-}
-
-fn trim_trailing_zeros(input: &[u8]) -> &[u8] {
-    let end = input
-        .iter()
-        .rposition(|&byte| byte != 0)
-        .map_or(0, |pos| pos + 1);
-    &input[..end]
+    if on_ramp.len() > 64 {
+        return false;
+    }
+    let prefix_matches = |long: &[u8], short: &[u8]| &long[0..short.len()] == short;
+    let suffix_is_zero = |a: &[u8], start: usize| a[start..].iter().all(|b| *b == 0);
```

Comment:
can you add a comment here explaining this two lines?

Reply:
🤔 I can't think of how I'd write those comments in a way that doesn't simply reiterate what the code says. If you think the code is unclear, that's fair: I can try to rewrite it in a less dense way rather than try to back it up with a comment.